app/ide-desktop/lib/dashboard/src/authentication/src/authentication/cognito.ts

```diff
@@ -320,3 +383,66 @@ const intoConfirmSignUpErrorOrThrow = (error: AmplifyError): ConfirmSignUpError
 
     throw error
 }
+
+// ========================
+// === SignInWithGoogle ===
+// ========================
+
+const signInWithGoogle = async (customState?: string) =>
+    amplify.Auth.federatedSignIn({
+        provider: amplify.CognitoHostedUIIdentityProvider.Google,
+        customState,
+    })
+        /** We don't care about the details in the success case, just that it happened. */
+        .then(() => null)
+
+// ========================
+// === SignInWithGoogle ===
+// ========================
+
+const signInWithGitHub = async () =>
```

Comment:
the same comment as above (please also apply in other places)

Reply:
All lambdas are now functions.